getting_started/workflow/best_practices/what_are_godot_classes.rst

```diff
@@ -1,80 +1,85 @@
+built-in classes.
 
-- constants
+Godot's internal classes have methods that register a class's data with a
+:ref:`ClassDB <class_ClassDB>`. This database provides runtime access to class
+information. ``ClassDB`` contains information about classes like:
 
+- properties
+- methods
+- constants
 - signals
 
-Furthermore, this ClassDB is what Objects actually check against when
-performing any operation. Access a property? Call a method? Emit a signal?
-It will check the database's records (and the records of the Object's base
-types) to see if the Object supports the operation. Every C++ Object defines
-a static `_bind_methods()` function that describes what C++ content it
-registers to the database and how.
-
-So, if the engine provides all of this data at startup, then how does
-a user define their own data? It'd be nice if users could define a custom
-set of data to be appended to an object's data. That way, users could inject
-their own properties and methods into the engine's Object query requests.
-
-*This* is what a :ref:`Script <class_Script>` is. Objects check their attached
-script before the database, so scripts can even override methods.
-If a script defines a `_get_property_list()` method, that data is appended to
-the list of properties the Object fetches from the ClassDB. The same holds
-true for other declarative code.
-
-This can lead to some users' confusion when they see a script as being
-a class unto itself. In reality, the engine just auto-instantiates the
-base engine class and then adds the script to that object. This then allows
-the Object to defer to the Script's content where the engine logic deems
-appropriate.
-
-A problem does present itself though. As the size of Objects increases,
-the scripts' necessary size to create them grows much, much larger.
-Creating node hierarchies demonstrates this. Each individual Node's logic
-could be several hundred lines of code in length.
-
-let's see a simple example of creating a single Node as a child.
+This ``ClassDB`` is what Objects check against when performing an operation like
+accessing a property or calling a method. ``ClassDB`` checks the database's
+records and the records of the Object's base types to see if the Object supports
+the operation.
+
+On the engine's side, every class defines a static `_bind_methods()` function
+that describes what C++ content it registers to the database and how. When you
+use the engine, you can extend the methods, properties, signals available from
+the ``ClassDB`` by attaching a :ref:`Script <class_Script>` to your node.
+
+Objects check their attached script before the database. This is why scripts can
+override built-in methods. If a script defines a `_get_property_list()` method,
+Godot appends that data to the list of properties the Object fetches from the
+ClassDB. The same is true for other declarative code.
+
+Scripts are not classes in themselves. Even scripts that don't inherit from a
```

Comment:
I think I understand what the second sentence intends to convey, but it might be a bit confusing. IIRC a script without `extends SomeClass` actually behaves the same as if it had `extends Reference`, which is implicit. So in itself a script cannot not inherit a built-tin type, it's always an `Object` (`Reference` if unspecified, otherwise most likely `Node` or more specific, or a `Resource` type).

Reply:
Confirmed that a script without `extends` inherits `Reference` implicitly:
![Screenshot_20190717_085320](https://user-images.githubusercontent.com/4701338/61353545-5e18d500-a870-11e9-94d7-d8f724a36373.png)
Which also means that it can't be attached to a `Node`, but it can be instanced from code and used as a `Reference`.